### PR TITLE
Migrate to Dart null safety system

### LINKED ISSUE #130: Migrate to sound null safety
Will you migrate to sound null safety?

The only dependency you have: [ffi](https://pub.dev/packages/ffi/versions/0.3.0-nullsafety.1) supports it.
Do you need help for that?

### PR DESCRIPTION
**Pull request by orevial**
Hi,

I propose this pull request to [migrate this plugin to null safety](https://dart.dev/null-safety/migration-guide) as it is the way to go now that Flutter 2 has been released.

Here are the changes I introduced:

* Update SDK dependency version to 2.12.0+
* Update `ffi` dependency to the new null-safe version (1.0.0)
* Run `dart migrate` on existing code to remove unsafe nulls, there is not so many changes to the codebase as your code was already pretty much null-safe, thank you for that 😄  🎉 
* Format some code

Thank you for the great work on this plugin and let me know if I can rework something !

Notes:

* This PR closes #130 
* This PR also updates deprecated codes after ffi lib update just as PR #134 did.

cc @crazecoder

### CODE REVIEW

**Comment by jlubeck** [comment]
Maybe @dcharkes can review?

File: lib/src/plaform/linux.dart [hunk 1]
```diff
@@ -14,10 +14,10 @@ int system(String command) {
   final systemP = dylib.lookupFunction<SystemC, SystemDart>('system');
 
   // Allocate a pointer to a Utf8 array containing our command.
-  final cmdP = Utf8.toUtf8(command);
+  final cmdP = command.toNativeUtf8();
 
   // Invoke the command, and free the pointer.
-  int result = systemP(cmdP);
+  final result = systemP(cmdP);
 //  cmdP.free();
```
Comment (dcharkes) [inline lib/src/plaform/linux.dart:22]: `calloc.free(cmdP);` otherwise you leak calloc'ed memory.

Same for the other files.

**Review by dcharkes — APPROVED** [review]
lgtm

File: lib/src/plaform/linux.dart (hunk 1 above)
Comment (orevial) [reply lib/src/plaform/linux.dart:22]: done

**Comment by mo-ah-dawood** [comment]
i tried to use this pr but got error

`UTF.toUtf8(...)` is removed  from  ffi 1.0.0
you have to change `command.toNativeUtf8()`

**Comment by orevial** [comment]
> i tried to use this pr but got error
> 
> `UTF.toUtf8(...)` is removed from ffi 1.0.0
> you have to change `command.toNativeUtf8()`

Hi @mo-ah-dawood, as far as I know I did change every `UTF.toUtf8(command)` to their `command.toNativeUtf8(...)` (or the Utf16 equivalent), are you sure you've been testing the PR ?

**Pull request merged**